snapcraft/internal/cache/_cache.py

```diff
@@ -44,3 +44,11 @@ def __init__(self, *, project_name):
         super().__init__()
         self.project_cache_root = os.path.join(
             self.cache_root, 'projects', project_name)
+
+
+class SnapcraftStagePackageCache(SnapcraftCache):
```

Comment:
At first I thought, is this going to be reusable? if not just implement this in `_apt`. Now I think you are forward thinking :smile: